actorguidelines.md

```diff
@@ -0,0 +1,84 @@
+# Actor Guidelines
+
+## Repository Structure
+
+Currently existing actors are stored in a common repository on Github: https://github.com/leapp-to/leapp-actors
+
+Such repository contains actors and schemas used by Leapp project. Actors stored on such repository should be stored following a directory structure based on their functionalities. As an example of such directories we have:
+* Common: Contains actors that provide functionalities that are used by a group of other actors.
+* Containerization: Contains actors specific for all types of migration to containers.
+* Containerization/Common: Contains actors with common functionalities used by other containerization actors.
+* Upgrade: Contains actors specific for all types of system-wide OS upgrades or migrations to the new OS version.
+* Upgrade/Common: Contains actors with common functionalities used by other upgrade actors.
+
+These is a structure that supports all existing use cases but, if necessary, it can be expanded in the future.
+
+
+## Actor Structure
+
+Each actor should be stored in a single directory with name based on the actor name (respecting naming conventions described below). An actor is composed of the following files:
+
+**_actor.yaml**: YAML file containing actor description used by Snactor library. This file will define used input/output channels and point to the script that will be executed by such actor. Even if not mandatory, a description should be provided in such file, in a way that developers can use it to understand purpose of an actor and how it is expected to work
```

Comment:
As @pirat89 mentioned, and I agree, the name of the script should have one specific hardcoded name we come up with. Without extension. In the Preupgrade Assistant, it was `check`, but that is not generic enough for Leapp. Maybe `exec`?

Reply:
Thanks for reviewing. I see your point and do agree that it can improve to have a pattern between all actors. But, personally, I am against removing ext code from source code files. Specially in a scenario where you can have Python/Bash/some other thing actors. Also, on current implementation of the framework, you need to define script name on YAML file, there is no hard coded default.  For sure, we can define a default for our project and update the guidelines. But, again personally, I would prefer let such decision for the developer. No strong opinions, only personal preference. I tried to update the code, let me knows if this can be a middle term between both approaches.

Reply:
Ok, let's leave it like that for now. I don't have such a strong opinion about that too. I guess QE may weigh in as they did for the PA in similar scenario, but we'll see.